### PR TITLE
Feature/issue42 scale up nozzle

### PR DESCRIPTION
**Pull request by chenziliang**
Make the nozzle be able to scale up. For our testing, it will boost performance a lot for single nozzle. We still can deploy multiple nozzle in parallel to consume one subscription to scale out. Scale up single nozzle has the benefits of absorb peak log volume to prevent data loss

### CODE REVIEW

File: drain/logging_splunk.go [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 import (
-	"code.cloudfoundry.org/lager"
 	"fmt"
-	"github.com/cloudfoundry-community/splunk-firehose-nozzle/splunk"
 	"math"
 	"strconv"
 	"strings"
 	"time"
+
+	"code.cloudfoundry.org/lager"
```
Comment (chenziliang) [inline drain/logging_splunk.go]: goimport formats the code

File: drain/logging_splunk.go [hunk 2]
```diff
@@ -18,23 +19,25 @@ type LoggingConfig struct {
...
+		clients: splunkClients,
+		config:  config,
+		events:  make(chan map[string]interface{}, config.QueueSize),
 	}
 }
 
 func (l *LoggingSplunk) Connect() bool {
-	go l.consume()
+	for _, client := range l.clients {
+		go l.consume(client)
```
Comment (chenziliang) [inline drain/logging_splunk.go]: Each goroutine uses its own client to avoid race and perf issue

File: drain/logging_splunk.go [hunk 3]
```diff
@@ -54,25 +57,25 @@ func (l *LoggingSplunk) consume() {
...
-func (l *LoggingSplunk) indexEvents(batch []map[string]interface{}) []map[string]interface{} {
+func (l *LoggingSplunk) indexEvents(client splunk.SplunkClient, batch []map[string]interface{}) []map[string]interface{} {
 	if len(batch) == 0 {
 		return batch
 	}
 	var err error
 	for i := 0; i < l.config.Retries; i++ {
-		l.logger.Info(fmt.Sprintf("Posting %d events", len(batch)))
-		err = l.client.Post(batch)
+		// l.logger.Info(fmt.Sprintf("Posting %d events", len(batch)))
```
Comment (chenziliang) [inline drain/logging_splunk.go]: We will do more tracing later by introducing a counter.

File: main.go [hunk 4]
```diff
@@ -106,15 +108,25 @@ func main() {
 		loggingClient = &drain.LoggingStd{}
 	} else {
 		splunkCLient := splunk.NewSplunkClient(*splunkToken, *splunkHost, *splunkIndex, parsedExtraFields, *skipSSL, logger)
-		loggingClient = drain.NewLoggingSplunk(logger, splunkCLient, loggingConfig)
 		logger.RegisterSink(sink.NewSplunkSink(*jobName, *jobIndex, *jobHost, splunkCLient))
+
+		var splunkClients []splunk.SplunkClient
+		for i := 0; i < *hecWorkers; i++ {
```
Comment (chenziliang) [inline main.go]: Create clients for different go routines

**Comment by luckyj5** [comment]
Looks good. Reviewed with @cholick!

**Pull request merged**